hydromt/raster.py

```diff
@@ -2247,6 +2248,245 @@ def tile_window(shape, px):
         with open(join(root, f"{mName}.yml"), "w") as f:
             yaml.dump({mName: yml}, f, default_flow_style=False, sort_keys=False)
 
+    def to_osm_tiles(
+        self,
+        root: str,
+        zoom_levels: list,
+        bbox: tuple = (),
+        method: str = "nearest",
+        min_lvl: int = 5,
+        driver="GTiff",
+        **kwargs,
+    ):
+        """Export rasterdataset to tiles in an osm structure.
+
+        Parameters
+        ----------
+        root : str
+            Path where the database will be saved
+            Database yml will be put one directory above
+        zoom_levels : list
+            Zoom levels to be put in the database
+        bbox : tuple, optional
+            A region where the tiles should be generated for
+            (if absent, the entire input raster dataset will be tiled)
+        method : str, optional
+            How to resample the data when downscaling.
+            E.g. 'nearest' for resampling with the nearest value
+        min_lvl : int, optional
+            Minimal starting level from to where to downscale from.
+            Increase if memory issues are encountered.
+        driver : str, optional
+            GDAL driver (e.g., 'GTiff' for geotif files), or 'netcdf4' for netcdf files.
+        **kwargs
+            Key-word arguments to write raster files
+        """
+        resampling = getattr(Resampling, method, None)
+        if resampling is None:
+            raise ValueError(f"Resampling method unknown: {method}.")
+        px_size = 256  # Fixed resolution for osm
+
+        assert self._obj.ndim == 2, "Only 2d datasets are accepted..."
```

Comment:
Either this should be removed, or we should raise a proper `ValueException` I think

Reply:
No, it checks whether the resampling method is available in rasterio. A ValueError (as far as I know) would then be correct.

Reply:
sorry I think I messed up the selection there. I was referring to the `assert` statement

Reply:
Maybe, we want to assert that there are indeed two dimensions, 3d data is not accepted. Don't know if that then should be a ValueError...

Reply:
Done